src/components/fields/money-field/money-field.js

```diff
@@ -0,0 +1,149 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import has from 'lodash.has';
+import requiredIf from 'react-required-if';
+import Constraints from '../../constraints';
+import Spacings from '../../spacings';
+import FieldLabel from '../../field-label';
+import MoneyInput from '../../inputs/money-input';
+import createSequentialId from '../../../utils/create-sequential-id';
+import FieldErrors from '../../field-errors';
+import filterDataAttributes from '../../../utils/filter-data-attributes';
+
+const sequentialId = createSequentialId('text-field-');
+
+const hasErrors = errors => errors && Object.values(errors).some(Boolean);
+
+class MoneyField extends React.Component {
+  static displayName = 'MoneyField';
+
+  static getAmountInputId = MoneyInput.getAmountInputId;
+  static getCurrencyDropdownId = MoneyInput.getCurrencyDropdownId;
+  static convertToMoneyValue = MoneyInput.convertToMoneyValue;
+  static parseMoneyValue = MoneyInput.parseMoneyValue;
+  static isEmpty = MoneyInput.isEmpty;
+  static isHighPrecision = MoneyInput.isHighPrecision;
+  static isTouched = MoneyInput.isTouched;
+
+  static propTypes = {
+    // MoneyField
+    id: PropTypes.string,
+    horizontalConstraint: PropTypes.oneOf(['xs', 's', 'm', 'l', 'xl', 'scale']),
+    errors: PropTypes.shape({
+      missing: PropTypes.bool,
+    }),
+    renderError: PropTypes.func,
+    isRequired: PropTypes.bool,
+    touched: PropTypes.shape({
```

Comment:
On a general note, I like the is/has/can prefix convention for variables, but it's not as useful for component props. The components will emit prop-type warnings anyways in case a wrong thing gets passed in.

Further, it looks nicer when rendering components without providing the values:

```diff
- <Button isDisabled={true} isReadOnly={true} />`
+ <Button disabled readonly />
```

This would obviously be a huge change and one that I would not do without a providing a codemod. I'm just mentioning it for now. I don't plan to take any action aside from maybe opening an issue for discussion.

Reply:
> I like the is/has/can prefix convention for variables, but it's not as useful for component props

I don't like it either but followed along for 16 months now. I think it should be a one of refactor for all components in UIKit?

Reply:
Yep, that's what I have in mind.